netebpfext/net_ebpf_ext_hook_provider.h

```diff
@@ -131,62 +152,52 @@ typedef struct _net_ebpf_extension_hook_provider_parameters
 NTSTATUS
 net_ebpf_extension_hook_provider_register(
     _In_ const net_ebpf_extension_hook_provider_parameters_t* parameters,
-    _In_ const net_ebpf_extension_hook_on_client_attach attach_callback,
-    _In_ const net_ebpf_extension_hook_on_client_detach detach_callback,
+    _In_ const net_ebpf_extension_hook_provider_dispatch_table_t* dispatch,
+    net_ebpf_extension_hook_attach_capability_t attach_capability,
     _In_opt_ const void* custom_data,
     _Outptr_ net_ebpf_extension_hook_provider_t** provider_context);
 
 /**
- * @brief Invoke the eBPF program attached to this hook. This must be called
- * inside a net_ebpf_extension_hook_client_enter_rundown/net_ebpf_extension_hook_client_leave_rundown block.
+ * @brief When hook provider supports multiple programs per hook, this callback function is invoked after
+ * every program invocation to determine whether the next program in the list should be invoked.
+ *
+ * @param[in] program_context Pointer to context passed to the eBPF program.
+ * @param[in] progrma_verdict Result returned by the eBPF program.
  *
- * @param[in] client Pointer to Hook NPI Client (a.k.a. eBPF Link object).
- * @param[in] context Context to pass to eBPF program.
- * @param[out] result Return value from the eBPF program.
- * @retval EBPF_SUCCESS The operation was successful.
- * @retval EBPF_NO_MEMORY Unable to allocate resources for this
- * operation.
+ * @returns TRUE if the next program should be invoked, FALSE otherwise.
  */
-_Must_inspect_result_ ebpf_result_t
-net_ebpf_extension_hook_invoke_program(
-    _In_ const net_ebpf_extension_hook_client_t* client, _Inout_ void* context, _Out_ uint32_t* result);
+typedef bool (*net_ebpf_extension_hook_process_verdict)(_Inout_ void* program_context, int program_verdict);
+
+net_ebpf_extension_hook_client_t*
```

Comment:
add doxygen

Reply:
This function was no longer needed. Removed this.